[Guided onboarding] Disable guided onboarding

alisonelizabeth opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/147644

This reverts some of the changes made in https://github.com/elastic/kibana/pull/143354, https://github.com/elastic/kibana/pull/143765, and https://github.com/elastic/kibana/pull/143837.

Tasks:

- [x] Add config to disable guided onboarding plugin
- [x] Remove "Setup guides" button from home page
- [x] Remove "Setup guides" link from help menu
- [x] Remove getting started route from home plugin

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/platform-onboarding (Team:Journey/Onboarding)

yuliacech commented on src/plugins/guided_onboarding/public/plugin.tsx:
```diff
@@ -41,12 +44,13 @@ export class GuidedOnboardingPlugin
+    const { ui: isGuidedOnboardingUiEnabled } = this.ctx.config.get<ClientConfigType>();
 
     // Initialize services
     apiService.setup(http, !!cloud?.isCloudEnabled);
```
I think we can add the config boolean to the setup and prevent any requests to be sent to the API if guided onboarding is disabled. Requests might be sent on the pages that check for guided onboarding state for their specific guide step, for example Rules in Security.
```suggestion
    apiService.setup(http, !!cloud?.isCloudEnabled && isGuidedOnboardingUiEnabled);
```

alisonelizabeth replied:
👍 good idea

yuliacech reviewed (commented):
Thanks a lot for working on this, @alisonelizabeth! 
I've tested locally and I think we need to delete this part
```
if (isCloudEnabled) {
          application.navigateToUrl('./home#/getting_started');
          return null;
        }
```
from the file `src/plugins/home/public/application/components/home.tsx` to prevent the page being stuck a redirect loop on the first visit. 
Also the button `Add integrations` on the home page needs to be converted back to `primary`. https://github.com/elastic/kibana/blob/836387c61457c1666b274ec7d6a4eab4f363a3bc/src/plugins/home/public/application/components/add_data/add_data.tsx#L78
Otherwise, I think you have everything covered 👍

alisonelizabeth commented:
Thanks for the review @yuliacech! Good catch on the additional changes needed to the `home` plugin. I believe I addressed your feedback. Mind taking another look?

yuliacech reviewed (approved):
Thanks for adding my feedback, @alisonelizabeth! 
For `apiService` I think we could probably just leave it with 1 boolean value to control if it's enabled or not (see this [commit](https://github.com/elastic/kibana/pull/147678/commits/0578fb831bc4fb54509a001fd7ee211ece2358f0) in 8.7 [PR](https://github.com/elastic/kibana/pull/147678)), but I don't mind having 2 values either.

alisonelizabeth commented:
Verified on cloud deployment

<img width="1590" alt="Screen Shot 2022-12-16 at 10 08 18 AM" src="https://user-images.githubusercontent.com/5226211/208128150-70244127-dde7-4927-a13b-8ca9e3fb084d.png">
<img width="307" alt="Screen Shot 2022-12-16 at 10 08 24 AM" src="https://user-images.githubusercontent.com/5226211/208128154-2f02d57d-1b98-4946-af55-707ef2e28d4c.png">
